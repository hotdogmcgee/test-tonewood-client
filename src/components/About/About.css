#About {
  text-align: center;
}

#About h2 {
  font-size: 25px;
  font-weight: 500;
}

#About a {
  text-decoration: none !important;
  color: #8899cc;
}

.modal {
  transition-timing-function: ease-in;
  transition-delay: 1s;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);
  z-index: 2;
}

/* Modal text can look a bit blurry. */
.modal-main {
  /* font-size: .8em; */

  font-size: 16px;
  border-radius: 10px;
  padding: 10px;
  position: fixed;
  background: white;
  width: 80%;
  height: auto;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  max-width: 400px;
}

.credentials-modal {
  text-align: center;
}

.display-block {
  display: block;
}

.display-none {
  display: none;
}

.button-container {
  text-align: center;
  display: flex;
  justify-content: center;
  margin-top: 10px;
  width: 100%;
}

.modal-button {
  width: 50%;
  height: 100%;
  padding: 5px 0px;
  font-size: 18px;
}
