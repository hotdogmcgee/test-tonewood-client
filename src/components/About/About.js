import React from 'react'
import './About.css'
import pdf from './How-to.pdf'

export default class About extends React.Component {

    renderAbout() {
        return (
            <>
            <h2>Welcome to the tonewood data project, here's how to get started.</h2>
            <p>The tonewood data project brings together luthiers worldwide to make it easy to find the properties of many tonewoods.</p>
            {/* <a href="https://www.pdfhost.net/index.php?Action=Download&File=1c27775b105ac4a5c744a876027023cc" target="blank">Link to PDF Instructions</a> */}
            <a href={pdf} download>Link to PDF Instructions</a>
            </>
        )
    }
    render() {
        return (
            this.renderAbout()
        )
    }
}