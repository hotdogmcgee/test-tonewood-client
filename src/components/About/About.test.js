import React from "react";
import About from "./About.js";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("About component", () => {
  it("renders an About component by default", () => {
    const wrapper = shallow(<About />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  //consider adding testing for clicks
  it("opens a Modal on button click", () => {
    const wrapper = shallow(<About />);

    wrapper.find(".modal-button").simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
