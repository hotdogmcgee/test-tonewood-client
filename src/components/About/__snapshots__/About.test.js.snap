// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component opens a Modal on button click 1`] = `
<Fragment>
  <h2>
    Welcome to the tonewood data project, here's how to get started.
  </h2>
  <p>
    The tonewood data project brings together luthiers worldwide to make it easy to find the properties of many tonewoods.
  </p>
  <a
    download="TW Data Project How-to"
    href="How-to.pdf"
  >
    Download PDF Instructions
  </a>
  <div
    className="button-container"
  >
    <Button
      className="modal-button"
      onClick={[Function]}
      type="button"
    >
      Show credentials to test Login
    </Button>
  </div>
  <CredentialsModal
    handleClose={[Function]}
    show={true}
  />
</Fragment>
`;

exports[`About component renders an About component by default 1`] = `
<Fragment>
  <h2>
    Welcome to the tonewood data project, here's how to get started.
  </h2>
  <p>
    The tonewood data project brings together luthiers worldwide to make it easy to find the properties of many tonewoods.
  </p>
  <a
    download="TW Data Project How-to"
    href="How-to.pdf"
  >
    Download PDF Instructions
  </a>
  <div
    className="button-container"
  >
    <Button
      className="modal-button"
      onClick={[Function]}
      type="button"
    >
      Show credentials to test Login
    </Button>
  </div>
  <CredentialsModal
    handleClose={[Function]}
    show={false}
  />
</Fragment>
`;
